content/articles/transaction-management-in-database/index.md.md

```diff
@@ -0,0 +1,220 @@
+CREATE TABLE STUDENTS(Id integer PRIMARY KEY, Name text, Regno varchar, Course text, Fees integer);
+
+
+INSERT INTO STUDENTS VALUES(1,'John', 'V20/680', 'Education', 40000);
+INSERT INTO STUDENTS VALUES(2,'Ahmed', 'G4/270', 'Computer Science',50000);
+INSERT INTO STUDENTS VALUES(3,'Ian', 'A43/476', 'Engineering', 62500);
+INSERT INTO STUDENTS VALUES(4,'Jane', 'A19/520', 'IT', 41200);
+INSERT INTO STUDENTS VALUES(5,'Robert','P62/491', 'Nursing',43700);
+INSERT INTO STUDENTS VALUES(6,'Hellen', 'V20/470', 'Education', 22000);
+INSERT INTO STUDENTS VALUES(7,'Joan','P62/801', 'Nursing',18000);
+COMMIT;
+
+
+SELECT * FROM STUDENTS;
+```
+Below is the output of the program.
+
+|ID | NAME   | REG NO  | COURSE     | FEES  |
+|-- | :-----:| :------:| :---------:| -----:|
+|1  | JOHN   |V20/630|Education|40000|
+|2  | AHMED  |G4/270|Computer science|50000|
+|3  | IAN    |A43/476|Engineering|62500|
+|4  | JANE|A19/520|IT|41200|
+|5  | ROBERT |P62/491|Nursing|43700|
+|6  | HELLEN |V20/470|Education|22000|
+|7  | JOAN   |P62/801|Nursing|18000|
+
+
+### Transactional control commands
+The Data Manipulation Language (DML) instructions are utilized with the transactional commands such as INSERT, UPDATE, and DELETE. These commands cannot be used with the Data Definition Language because they are automatically committed to the database. Below are the transaction commands; 
+
+1. COMMIT- Saves the changes done against the database.
+2. ROLLBACK- Redoes the changes done before being commited to the database.
+3. SAVEPOINT-saves a point in a transaction on which to carry out a rollback.
+4. SET TRANSACTION- Gives a name to a transaction.
+
+### COMMIT command
+The COMMIT command saves changes made to a database as part of a transaction. The COMMIT command preserves all database transactions since the previous COMMIT or ROLLBACK command.
+The syntax of the COMMIT command; 
+
+```SQL
+COMMIT
+```
+We can delete the records from the table that have fees=40000 and commit the changes in the database.
+
+```SQL
+DELETE FROM STUDENTS
+WHERE FEES = 40000;
+COMMIT;
+```
+This command will delete the row from the table which has fees equal to 40000 and save the changes to the database.
+When this command is executed the results are as follows;
+
+Output;
+|ID | NAME   | REG NO  | COURSE     | FEES  |
+|-- | :-----:| :------:| :---------:| -----:|
+|2  | AHMED  |G4/270|Computer science|50000|
+|3  | IAN    |A43/476|Engineering|62500|
+|4  | JANE|A19/520|IT|41200|
+|5  | ROBERT |P62/491|Nursing|43700|
+|6  | HELLEN |V20/470|Education|22000|
+|7  | JOAN   |P62/801|Nursing|18000|
+
+The first row is deleted and the results saved.
+
+### ROLLBACK command
+The ROLLBACK command can only be used to undo transactions from the last ROLLBACK or COMMIT command issued.
+The ROLLBACK command syntax is as follows;
+
+```SQL
+ROLLBACK;
+```
+
+Consider deleting a record from the table where fees =40000 and ROLLBACK the changes;
+
+```SQL
+DELETE FROM STUDENTS
+WHERE FEES = 40000;
+ROLLBACK;
```

Comment:
Running this does not result in the output you say it will on my end. 
It also raises the same `there is no transaction in progress` warning
```bash
mega=# SELECT * FROM STUDENTS;
 id |  name  |  regno  |      course      | fees
----+--------+---------+------------------+-------
  1 | John   | V20/680 | Education        | 40000
  2 | Ahmed  | G4/270  | Computer Science | 50000
  3 | Ian    | A43/476 | Engineering      | 62500
  4 | Jane   | A19/520 | IT               | 41200
  5 | Robert | P62/491 | Nursing          | 43700
  6 | Hellen | V20/470 | Education        | 22000
  7 | Joan   | P62/801 | Nursing          | 18000
(7 rows)

mega=# DELETE FROM STUDENTS
mega-# WHERE FEES = 40000;
DELETE 1
mega=# ROLLBACK;
WARNING:  there is no transaction in progress
ROLLBACK
mega=# SELECT * FROM STUDENTS;
 id |  name  |  regno  |      course      | fees
----+--------+---------+------------------+-------
  2 | Ahmed  | G4/270  | Computer Science | 50000
  3 | Ian    | A43/476 | Engineering      | 62500
  4 | Jane   | A19/520 | IT               | 41200
  5 | Robert | P62/491 | Nursing          | 43700
  6 | Hellen | V20/470 | Education        | 22000
  7 | Joan   | P62/801 | Nursing          | 18000
(6 rows)
```